crates/net/network/src/peers/manager.rs

```diff
@@ -283,16 +283,23 @@ impl PeersManager {
         self.connection_info.decr_pending_in();
         self.connection_info.inc_in();
 
+        // check weather if the peer is trustable or not
+        let mut is_trusted = self.trusted_peer_ids.contains(&peer_id);
+        if self.trusted_nodes_only && !is_trusted {
+            self.queued_actions.push_back(PeerAction::DisconnectUntrustedIncoming { peer_id });
+            return
+        }
+
         match self.peers.entry(peer_id) {
             Entry::Occupied(mut entry) => {
-                let value = entry.get_mut();
-                if value.is_banned() {
+                let peer = entry.get_mut();
+                if peer.is_banned() {
                     self.queued_actions.push_back(PeerAction::DisconnectBannedIncoming { peer_id });
                     return
                 }
-                value.state = PeerConnectionState::In;
+                peer.state = PeerConnectionState::In;
 
-                let is_trusted = value.is_trusted() || self.trusted_peer_ids.contains(&peer_id);
+                is_trusted = peer.is_trusted() || is_trusted;
```

Comment:
just a nit: `is_trusted = is_trusted || peer.is_trusted()` maybe a better way.

If `is_trusted` is `true` previously, it could short circuit calling functions `is_trusted()`, although it's inline :)

Reply:
thanks for the advice